halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/ResourceConfig.java

```diff
@@ -22,24 +22,27 @@
 import org.springframework.stereotype.Component;
 import org.yaml.snakeyaml.DumperOptions;
 import org.yaml.snakeyaml.Yaml;
-import org.yaml.snakeyaml.introspector.Property;
-import org.yaml.snakeyaml.nodes.NodeTuple;
-import org.yaml.snakeyaml.nodes.Tag;
-import org.yaml.snakeyaml.representer.Representer;
+
+import java.io.File;
 
 @Component
 public class ResourceConfig {
   /**
-   * Path to where the halconfig file is located.
+   * Directory containing the halconfig.
    *
-   * @param path Defaults to ~/.hal/config.
+   * @param path Defaults to "~/.hal".
    * @return The path with home (~) expanded.
    */
   @Bean
-  String halconfigPath(@Value("${halconfig.directory.halconfig:~/.hal/config}") String path) {
+  String halconfigDirectory(@Value("${halconfig.directory.halconfig:~/.hal}") String path) {
     return normalizePath(path);
   }
 
+  @Bean
+  String halconfigPath(@Value("${halconfig.directory.halconfig:~/.hal}") String path) {
+    return normalizePath(path + File.separator + "config");
```

Comment:
Can live with this, but you may want to use a combination of the `File` constructors and `File.getCanonicalPath()` to avoid doing the string manipulation yourself.